File: leetcode/easy/1844. Replace All Digits with Characters/solution.js
// https://leetcode.com/problems/replace-all-digits-with-characters/
/**
 * @param {string} s
 * @return {string}
 */
function replaceDigits(s) {
    let res = '';
    for (let i = 0; i < s.length; i += 2) {
        const char = s[i], num = parseInt(s[i+1], 10);
        // res += num ? ...
        // 이런식으로 작성하면 num이 0 일 경우 false 처리가 된다.
        // 따라서 type 체크를 바르게 하도록 해야한다.
        res += !isNaN(num) ? char + String.fromCharCode(char.charCodeAt(0) + num) : char;
    }
    console.log(res);
    return res;
};

replaceDigits('v0g6s4d');

File: leetcode/easy/1844. Replace All Digits with Characters/solution.cc
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    string replaceDigits(string s) {
        string str = s;
        for (int i = 0; i < s.size(); i += 2) {
            str[i+1] = s[i] + (s[i + 1] - '0');
        }
        return str;
    }
};

int main() {
    Solution solution;
    cout << solution.replaceDigits("v0g6s4d");
    return 0;
}
